common/configuration/db_provider.go

```diff
@@ -0,0 +1,30 @@
+package configuration
+
+import (
+	"context"
+	"net/url"
+)
+
+// DBProvider is a configuration provider that stores configuration in its key.
+type DBProvider struct {
+	DB bool `help:"Write configuration values to the database." group:"Provider:" xor:"configwriter"`
+}
+
+var _ MutableProvider[Configuration] = DBProvider{}
+
+func (DBProvider) Role() Configuration { return Configuration{} }
+func (DBProvider) Key() string         { return "db" }
+
+func (d DBProvider) Writer() bool { return d.DB }
+
+func (DBProvider) Load(ctx context.Context, ref Ref, key *url.URL) ([]byte, error) {
+	return []byte(key.Host), nil
```

Comment:
I'm a bit on the fence about whether it's a good idea to have the content encoded into the URL. Config values are arbitrary JSON, so they could be quite large. I think it might be preferable if the URL was just a reference, and the provider retrieves the value. It's an extra DB round trip, but I think that's okay.

Reply:
Yeah... what do you think about having the URL be "db://{{id}}" and then DBProvider.Load retrieves the config value by ID?

Reply:
Actually that wouldn't make sense in `DatabaseResolver.Set`, then, since neither the Ref nor the URL would encapsulate the actual config value we're storing... I'll go with the base64 encoding for now

Reply:
That could work, but then you'll need two queries - one for the resolver to retrieve the ID and one for the provider to load the value. It's probably a bit more efficient to make it something like `db://[<module>.]<name>` - you actually won't need to parse it at all because the resolver passes the Ref down to the provider, it's just for users.

Reply:
I don't think we should be base64 encoding anywhere.

Reply:
I set this up to use b64 for now. My only concern with doing something like `db://[module.]name` is the question I mentioned below, around what `DatabaseResolver.Set` should do in that case.

Reply:
fwiw I am also very uncomfortable encoding the value into the URL haha

Reply:
Done! URL is now just "db://" and all DB accesses except for `List` have been moved to the provider.